radix-engine-common/src/data/scrypto/custom_validation.rs

```diff
@@ -0,0 +1,185 @@
+use super::*;
+use crate::constants::*;
+use crate::types::{GlobalAddress, NodeId, PackageAddress};
+use crate::*;
+use sbor::rust::prelude::*;
+use sbor::traversal::TerminalValueRef;
+use sbor::*;
+
+impl ValidatableCustomTypeExtension<()> for ScryptoCustomTypeExtension {
+    fn apply_custom_type_validation<'de>(
+        _: &Self::CustomTypeValidation,
+        _: &TerminalValueRef<'de, Self::CustomTraversal>,
+        _: &mut (),
+    ) -> Result<(), ValidationError> {
+        Ok(())
+    }
+}
+
+/// A slightly duplicated model to break circular dependency.
```

Comment:
Minor: to fix this circular dependency this validation should be part of the system layer rather than `radix-engine-common`. Maybe update comment regarding this?

Reply:
Yeah, the implementation is part of the system layer. Here is just the interface.

I will add more comments.